agent/agent.go

```diff
@@ -2066,6 +2092,9 @@ func (a *Agent) removeServiceLocked(serviceID string, persist bool) error {
 		return fmt.Errorf("ServiceID missing")
 	}
 
+	// Shut down the config watch in the service manager.
+	a.serviceManager.RemoveService(serviceID)
+
```

Comment:
Does this need to be conditional on the Config and/or `a.serviceManager` being nil?